team_picker/routes/teams.js

```diff
@@ -0,0 +1,94 @@
+const express = require("express");
+const router = express.Router();
+const knex = require("../db/client");
+
+router.get('/', (req, res) => {
+  res.render('main'); 
+});
+
+router.get("/new", (req, res) => {
+  res.render("new");
+});
+
+// Enter new cohort
+router.post("/new", (req, res) => {
+  console.log("Saving data to database from new teams")
+  knex("cohorts")
+    .insert({
+      name: req.body.name,
+      logoUrl: req.body.logoUrl,
+      members: req.body.members
+    })
+    .returning("*")
+    .then(cohorts => {
+      res.render('show', { cohort: cohorts[0], pickerMethod: undefined, quantity: 0 });
```

Comment:
This should redirect to `/teams/cohorts[0].id` instead of rendering `show` directly. If you were to hit reload, on that page, it would actually create _another_ cohort with the same values. If you redirect, the user ends up on the updated URL (`/teams/X`) and they can reload as many times as they want. You don't want them to accidentally reload on the `POST` . Also, note that you're having to duplicate the logic to set `pickerMethod` and `quantity`, which is already handled [here](https://github.com/rosadiaz/homeworks/pull/6/files#diff-e79ea032b67d8a0bcdd34e679a2c241fR44).

Given the above, I believe you could then also change L22 to `returning('id')`, since you only really need the ID in order to do the redirect, you don't need the rest of the data in the row (if I'm understanding how `returning` works, which I've never used myself).

Looking at the [docs](https://knexjs.org/#Builder-insert), however, seems to be like the _default_ behaviour is that it will automatically resolve the promise with the ID of the inserted record. So I thought it _might_ be possible to do something like:
```es6
  knex("cohorts")
    .insert({...})
    .then( id => {
      res.redirect(`/teams/${id}`);
    });
```
but that doesn't seem to work... I'm either misunderstanding the docs, or they're outdated or something.  This does work:
```es6
  knex("cohorts")
    .insert({...})
    .returning('id')
    .then(id => {
      res.redirect(`/teams/${id}`);
    });

```

Reply:
I figured it out! This is how you can clean this up:
```es6
  knex("cohorts")
    .insert({...}, 'id')
    .then(id => {
      res.redirect(`/teams/${id}`);
    });
```
By passing a second argument to `insert` (in this case `'id'`), you can specify which column to return, and whose value is what is passed to the `then` callback